lib/recording/RecordingEntrySession.js

```diff
@@ -484,6 +502,25 @@ class RecordingEntrySession {
         this.logger = loggerBase.getLogger('RecordingEntrySession', '[' + this.recordingSessionId + '] ');
         return this.playlistGenerator.setBasePath(this.recordingSessionPath);
     }
+
+    isUnlimitedRecordingDurationEnabled() {
+        return backendClient.getEntryInfo(this.entryObject.entryId)
+            .then((entryObj) => {
+                if (entryObj.capabilities && entryObj.capabilities.indexOf('FEATURE_UNLIMITED_RECORDING_DURATION') >= 0) {
+                    this.logger.info(`Feature unlimited recording duration is enabled. Recording will be split`);
+                    return Q.resolve(true);
+                }
+                else {
+                    this.logger.info(`Recording will stop, as unlimited recording duration feature is disabled!!!`);
+                    this.disableRecording = true;
+                    return Q.resolve(false);
+                }
+            })
+            .catch((err)=> {
+                this.logger.error(`Failed to get entry info. Recording will stop, assuming unlimited recording duration feature is disabled. Error: ${ErrorUtils.error2string(err)}`);
+                return Q.resolve(false);
```

Comment:
Shouldn't you add here this.disableRecording = true;?

Reply:
yes. done